src/librustc_parse_format/lib.rs

```diff
@@ -243,11 +243,14 @@ impl<'a> Iterator for Parser<'a> {
                 _ => Some(String(self.string(pos))),
             }
         } else {
-            if self.is_literal && self.cur_line_start != self.input.len() {
+            if self.is_literal {
                 let start = self.to_span_index(self.cur_line_start);
                 let end = self.to_span_index(self.input.len());
-                self.line_spans.push(start.to(end));
-                self.cur_line_start = self.input.len();
+                let span = start.to(end);
+                if self.line_spans.last().map(|last_span| last_span.end) != Some(span.end) {
+                    self.line_spans.push(span);
+                    self.cur_line_start = self.input.len();
+                }
```

Comment:
```suggestion
                if self.line_spans.last() != Some(span) {
                    self.line_spans.push(span);
                }
```

Reply:
Done; confirmed that it seems to pass tests. I was concerned that this would cause a problem for other code using `cur_line_start`, but it looks like none of the other code using `cur_line_start` can run once this condition has happened.